### PR TITLE
[KYUUBI #3715] Remove unexpected mysql connector jar

### LINKED ISSUE #3715: [LICENSE] Remove unexpected mysql jdbc jar from repo and source release
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

It was added in https://github.com/apache/incubator-kyuubi/pull/3240, and I didn't notice a binary Jar was shipped in the patch, and the worse thing is it was included in the source release tarball.

We should 
1. remove it from the git repo ASAP, cc @zhaomin1423
2. and enhance our release process to avoid such things happen again

### Affects Version(s)

master/1.6

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes. I can submit a PR independently to fix.
- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Fix #3715.

It was added in https://github.com/apache/incubator-kyuubi/pull/3240, and I didn't notice a binary Jar was shipped in the patch, and the worse thing is it was included in the source release tarball.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: integration-tests/kyuubi-jdbc-it/pom.xml [hunk 1]
```diff
@@ -79,7 +89,51 @@
...
+                            <goal>build-classpath</goal>
+                        </goals>
+                        <configuration>
+                            <includeScope>test</includeScope>
+                            <outputProperty>test_classpath</outputProperty>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>copy-module-dependencies</id>
+                        <phase>process-test-resources</phase>
```
Comment (pan3793) [inline integration-tests/kyuubi-jdbc-it/pom.xml:125]: overwrite the phase of coping jars

**Comment by cfmcgrady** [comment]
The alternative approach is [here](https://github.com/cfmcgrady/incubator-kyuubi/pull/28).

**Review by zhaomin1423 — APPROVED** [review]
LGTM, thanks.

**Comment by zhaomin1423** [comment]
> The alternative approach is [here](https://github.com/cfmcgrady/incubator-kyuubi/pull/28).

+1, it's also good.

**Review by cxzl25 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master/1.6

**Pull request closed without merging**